src/main/java/spinbox/Parser.java

```diff
@@ -43,98 +45,164 @@ public static void setPageTrace(ArrayDeque<String> pageTraceNew) {
     }
 
     /**
-     * Builds the required page data for command input. Return page and maybe moduleCode
+     * Appends the page or page and "modules" from pageTrace.
+     * @param pageData The original empty page data.
+     * @param tempPageTrace The cloned version of the original ArrayDeque pageTrace.
+     * @param lastElement The last element of the ArrayDeque pageTrace.
+     * @return The full page data.
+     */
+    private static String emptyPageComponentAppender(String pageData, ArrayDeque<String> tempPageTrace,
+                                                     String lastElement) {
+        String fullPageData;
+        fullPageData = pageData.concat(lastElement);
+        tempPageTrace.removeLast();
+        if (tempPageTrace.size() != 0) {
+            String lastComponent = tempPageTrace.getLast();
+            String finalFullPageData;
+            finalFullPageData = fullPageData.concat(" " + lastComponent.toUpperCase());
+            LOGGER.info("Exiting static method emptyPageComponentAppender with pageData: " + finalFullPageData);
+            return finalFullPageData;
+        } else {
+            LOGGER.info("Exiting static method emptyPageComponentAppender with pageData: " + fullPageData);
+            return fullPageData;
+        }
+    }
+
+    /**
+     * Appends page from input and maybe module code from pageTrace to the full page data.
+     * @param pageData The original empty page data.
+     * @param firstPageComponent The first component of the split input page data.
+     * @param tempPageTrace The cloned version of the original ArrayDeque pageTrace.
+     * @return The full page data.
+     */
+    private static String singlePageComponentAppender(String pageData, String firstPageComponent,
+                                                      ArrayDeque<String> tempPageTrace) {
+        String fullPageData;
+        switch (firstPageComponent) {
+        case "main":
+        case "calendar":
+            fullPageData = pageData.concat(firstPageComponent);
+            break;
+        case "modules":
+            fullPageData = emptyPageComponentAppender(pageData, tempPageTrace, firstPageComponent);
+            break;
+        default:
+            // pageData is a module code
+            fullPageData = pageData.concat("modules " + firstPageComponent.toUpperCase());
+        }
+        LOGGER.info("Exiting static method singlePageComponentAppender with pageData: " + fullPageData);
```

Comment:
LOGGER.entering / LOGGER.exiting seems more appropriate here, unless these are more important to log

Reply:
Changed to the above methods.